lib/editor/util/validation.js

```diff
@@ -109,6 +124,12 @@ export function validate (
         return false
       }
     case 'TEXT':
+      if (name === 'stop_name' && isOptionalAndEmpty) {
+        if (!locationType || locationType <= 2) {
+          reason = 'Stop name is required for your current location type'
+          return {field: name, invalid: isOptionalAndEmpty, reason}
+        }
+      }
       if (name === 'route_short_name' && !value && entity && entity.route_long_name) {
```

Comment:
This is why we have the complexity eslint rule. The first two if statements should be combined into one and then the one with the route_short_name could be a continuation of this first one with an else if block. But even more so, there is this massive [switch statement code smell](https://refactoring.guru/smells/switch-statements) that ought to be refactored.